cmd/clusterctl/client/upgrade.go

```diff
@@ -30,6 +30,8 @@ import (
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
 )
 
+const upgradeItemProviderNameError = "invalid provider name %q. Provider name should be in the form namespace/provider:version or provider:version"
```

Comment:
Q: should we also add an example in upgrade_apply.go with the new syntax? (~ l.59)

Reply:
@sbueringer Great suggestion. Fixed! 

New output:
```
/bin/clusterctl upgrade apply -h
The upgrade apply command applies new versions of Cluster API providers as defined by clusterctl upgrade plan.

New version should be applied ensuring all the providers uses the same cluster API version
in order to guarantee the proper functioning of the management cluster.

Usage:
  clusterctl upgrade apply [flags]

Examples:
  # Upgrades all the providers in the management cluster to the latest version available which is compliant
  # to the v1alpha4 API Version of Cluster API (contract).
  clusterctl upgrade apply --contract v1alpha4
  
  # Upgrades only the aws provider to the v2.0.1 version.
  clusterctl upgrade apply --infrastructure aws:v2.0.1

Flags:
  -b, --bootstrap strings           Bootstrap providers instance and versions (e.g. kubeadm:v1.1.5) to upgrade to. This flag can be used as alternative to --contract.
      --contract string             The API Version of Cluster API (contract, e.g. v1alpha4) the management cluster should upgrade to
  -c, --control-plane strings       ControlPlane providers instance and versions (e.g. kubeadm:v1.1.5) to upgrade to. This flag can be used as alternative to --contract.
      --core string                 Core provider instance version (e.g. cluster-api:v1.1.5) to upgrade to. This flag can be used as alternative to --contract.
  -h, --help                        help for apply
  -i, --infrastructure strings      Infrastructure providers instance and versions (e.g. aws:v2.0.1) to upgrade to. This flag can be used as alternative to --contract.
      --ipam strings                IPAM providers and versions (e.g. infoblox:v0.0.1) to upgrade to. This flag can be used as alternative to --contract.
      --kubeconfig string           Path to the kubeconfig file to use for accessing the management cluster. If unspecified, default discovery rules apply.
      --kubeconfig-context string   Context to be used within the kubeconfig file. If empty, current context will be used.
      --runtime-extension strings   Runtime extension providers and versions (e.g. test:v0.0.1) to upgrade to. This flag can be used as alternative to --contract.
      --wait-provider-timeout int   Wait timeout per provider upgrade in seconds. This value is ignored if --wait-providers is false (default 300)
      --wait-providers              Wait for providers to be upgraded.

Global Flags:
      --config $HOME/.cluster-api/clusterctl.yaml   Path to clusterctl configuration (default is $HOME/.cluster-api/clusterctl.yaml) or to a remote location (i.e. https://example.com/clusterctl.yaml)
  -v, --v int                                       Set the log level verbosity. This overrides the CLUSTERCTL_LOG_LEVEL environment variable.

```

Reply:
Thank you very much!!

I'll leave the further review of the PR to Yuvaraj/Fabrizio.